src/ui/public/visualize/visualization_editor.js

```diff
@@ -36,7 +36,7 @@ uiModules
 
         $scope.$on('render', (event) => {
           event.preventDefault();
-          $scope.renderFunction();
+          $timeout(() => { $scope.renderFunction(); });
```

Comment:
Since we discussed this: I would leave this in for now, since it works. In general we try to refactor and improve the code within the visualize team and we might be able to remove this in an upcoming refactoring, but for now, I would leave it as is, i.e. with the timeout.